deep_qa/layers/wrappers/multi_input_time_distributed.py

```diff
@@ -0,0 +1,161 @@
+    to allow for inputs that are lists of tensors.  Currently, while masking is supported masking (i.e., if any of the inputs
+    has a mask) is not supported.
+
+    Inputs:
+        - probabilities, shape ``(batch, ..., N, ...)``,
+          with an optional mask of the same shape,
+          where N is the number of y's in the above equation
+          (i.e. the number of probabilities being combined in the product),
+          in the dimension corresponding to the specified axis.
+    Output:
+        - X, shape ``(batch, ..., ...)``, one less than input dims,
+          with an optional mask of the same shape.  The lost dimension corresponds to the specified axis.
+          The output mask is the result of `K.any()` on the input mask, along the specified axis.
+    Parameters
+    ----------
+    axis : int, default=-1
+        The axis over which to combine probabilities.
+    name : string, default='noisy_or'
+        Name of the layer, ued to debug both the layer and its parameter.
+    param_init : string, default='uniform'
+        The initialization of the noise parameter.
+    noise_param_constraint : Keras Constraint, default=None
+        Optional, a constraint which would be applied to the noise parameter.
+    """
+
+    @overrides
+    def build(self, input_shape):
+        num_inputs = len(input_shape)
+        input_specs = []
+        for i in range(num_inputs):
+            assert len(input_shape[i]) >= 3
+            input_specs.append(InputSpec(shape=input_shape[i]))
+        self.input_spec = input_specs
+        child_input_shapes = []
+        for i in range(num_inputs):
+            child_input_shape = (input_shape[i][0],) + input_shape[i][2:]
+            child_input_shapes.append(child_input_shape)
+        if not self.layer.built:
+            self.layer.build(child_input_shapes)
+            self.layer.built = True
+        super(TimeDistributed, self).build()
+
+    @overrides
+    def get_output_shape_for(self, input_shape):
+        if isinstance(input_shape, tuple):
+            child_input_shape = (input_shape[0],) + input_shape[2:]
+            child_output_shape = self.layer.get_output_shape_for(child_input_shape)
+            timesteps = input_shape[1]
+            return (child_output_shape[0], timesteps) + child_output_shape[1:]
+        elif isinstance(input_shape, list):
+            num_inputs = len(input_shape)
+            timesteps = input_shape[0][1]
+            output_shapes = []
+            for i in range(num_inputs):
+                child_input_shape = (input_shape[i][0],) + input_shape[i][2:]
+                child_output_shape = self.layer.get_output_shape_for(child_input_shape)
+                output_shapes.append((child_input_shape[0], timesteps) + child_output_shape[1:])
+            return output_shapes[0]
+        else:
+            raise Exception("Invalid input_shape type.")
+
+    @overrides
+    def compute_mask(self, x, input_mask=None):
```

Comment:
Actually, I think I would just recommend punting on this for now, because we should put the `call` and other code directly in our `TimeDistributed` class.  Then each of our subclasses that handle masks differently can incorporate the multi-input into their `compute_mask` methods.  The trouble with having a single implementation here is that different subclasses do different mask computations, so you can't really have a single general method.

If you want to include this in this PR, instead of punting on it, you can probably move what you have here to the `TimeDistributedWithMask` class.